File: solutions/Shift-to-Create-Range.cpp
bool solve(vector<int> &nums)
{
    int size = nums.size();
    int a = 0, b = 0;
    for (int i = 0; i < size; i++)
    {
        int diff = nums[(i + 1) % size] - nums[i];
        if (diff != 1)
            a++;
        if (diff != -1)
            b++;
    }

    return size == 1 || a == 1 || b == 1;
}

File: solutions/Shift-to-Create-Range.py
class Solution:
    def solve(self, nums):
        size = len(nums)
        if size == 1:
            return True

        diffs = set([nums[i] - nums[i - 1] for i in range(1, size)])
        return (
            diffs == {1}
            or diffs == {-1}
            or diffs == {1, -(size - 1)}
            or diffs == {-1, size - 1}
        )
